src/modinv64_impl.h

```diff
@@ -320,22 +320,28 @@ static void secp256k1_modinv64_update_de_62(secp256k1_modinv64_signed62 *d, secp
     /* Compute limb 1 of t*[d,e]+modulus*[md,me], and store it as output limb 0 (= down shift). */
     cd += (int128_t)u * d1 + (int128_t)v * e1;
     ce += (int128_t)q * d1 + (int128_t)r * e1;
-    cd += (int128_t)modinfo->modulus.v[1] * md;
-    ce += (int128_t)modinfo->modulus.v[1] * me;
+    if (modinfo->modulus.v[1]) { /* Optimize for the case where limb of modulus is zero. */
```

Comment:
Do we know if this is evaluated at compile-time? Maybe that's the reason why this is only sometimes faster.

Reply:
It's not in general (it is when modinv* is only invoked once in a binary, in which case it's inlined in its entirety).

I believe that the cost of the conditional jump here outweighs the cost of 2 wide multiplications.

It is hard to be actually sure here though, as our benchmarks may be priming the branch predictor in a way that isn't representative of what happens in a real world load.

Reply:
A while ago I was reading up on defenses against Spectre, and since it is based on branch mispredictions, one of the simplest defenses is of course to avoid branches in context where we process secret data, even if the branch is on public data. [1]  I think we have a few "interesting" branches in our code base, where I could see how we compute on secret data if the wrong branch is predicted. This branch is not obviously of that kind but I may be a good rule of thumb to avoid branches when in doubt.

This is more of a note, I'm not saying that this branch must be removed. I should provide more evidence, and it's on my list to open an issue about this and point to more interesting cases in our code. 

[1] See also https://fc21.ifca.ai/papers/18.pdf but this is different because they do automatic rewriting.